src/main/java/org/eclipsefoundation/marketplace/service/CachingService.java

```diff
@@ -0,0 +1,26 @@
+/* Copyright (c) 2019 Eclipse Foundation and others.
+ * This program and the accompanying materials are made available
+ * under the terms of the Eclipse Public License 2.0
+ * which is available at http://www.eclipse.org/legal/epl-v20.html,
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipsefoundation.marketplace.service;
+
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Interface defining the caching service to be used within the application.
+ * 
+ * @author Martin Lowe
+ */
+public interface CachingService {
+
+	Optional<Object> get(String key);
+
+	<T> void put(String key, T value);
```

Comment:
put is not required anymore if you follow my advice of leveraging the `Cache::get(K, Callable<? extends V>)`